packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -213,23 +315,33 @@ export class ModularEditBuilder extends ProgressiveEditBuilder<FieldChangeMap> {
         fieldKind: FieldKindIdentifier,
         change: FieldChangeset,
     ): void {
-        let fieldChangeMap: FieldChangeMap = new Map();
-        fieldChangeMap.set(field, { fieldKind, change });
+        let fieldChangeMap: FieldChangeMap = new Map([[
+            field,
+            { fieldKind, change },
+        ]]);
 
         let remainingPath = path;
         while (remainingPath !== undefined) {
             const editor = getChangeHandler(this.fieldKinds, remainingPath.parentFieldKind).editor;
-            const fieldChange = editor.buildChildChange(remainingPath.parentIndex, fieldChangeMap);
-            fieldChangeMap = new Map();
-            fieldChangeMap.set(
+            const nodeChange: NodeChangeset = { fieldChanges: fieldChangeMap };
+            const fieldChange = editor.buildChildChange(remainingPath.parentIndex, nodeChange);
+            fieldChangeMap = new Map([[
                 remainingPath.parentField,
                 { fieldKind: remainingPath.parentFieldKind, change: brand(fieldChange) },
-            );
+            ]]);
             remainingPath = remainingPath.parent;
         }
 
         this.applyChange(fieldChangeMap);
     }
+
+    setValue(path: UpPathWithFieldKinds, value: Value): void {
+        const valueChange: ValueChange = { value };
```

Comment:
We try to avoid having optional fields explicitly set to undefined.
```suggestion
        const valueChange: ValueChange = value === undefined ? {} : { value };
```